integration_test/tests/signer.rs

```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: CC0-1.0
+
+//! Tests for methods found under the `== Signer ==` section of the API docs.
+
+#![allow(non_snake_case)] // Test names intentionally use double underscore.
+#![allow(unused_imports)] // Because of feature gated tests.
+
+use integration_test::{Node, NodeExt as _, Wallet};
+use node::{mtype, Input, Output};
+use node::vtype::*;             // All the version specific types.
+
+#[test]
+#[cfg(not(feature = "v21_and_below"))]
+fn signer__enumerate_signers() {
+	let script_path = integration_test::random_tmp_file();
+	let script_body = "#!/bin/sh\necho '[{\"fingerprint\":\"deadbeef\",\"name\":\"TestSigner\"}]'\n";
+	std::fs::write(&script_path, script_body).expect("write signer script");
+    
+	#[cfg(unix)]
+	{
+		use std::os::unix::fs::PermissionsExt;
+		std::fs::set_permissions(&script_path, std::fs::Permissions::from_mode(0o755))
+			.expect("chmod");
+	}
```

Comment:
A few comments:

- This using tabs, can we use spaces please (tabs vs spaces is a age old religious war - I'm on team spaces).
-  Line 18 has trailing whitespace
- Perhaps add a code comment saying "Script needs to be executable" or some such thing
- Perhaps a code comment on the `script_body` - I have no idea why the script is as it is?
- Its not clear to me what happens if this test is run on non-unix machine. I don't actually care about non-unix machines though, maybe put the `cfg` on the whole test to make it clear we only support UNIX for this test.

Reply:
> * Its not clear to me what happens if this test is run on non-unix machine. I don't actually care about non-unix machines though, maybe put the `cfg` on the whole test to make it clear we only support UNIX for this test.

I feel like the tests are mainly for CI to check that any code changes or changes to the RPCs are ok, so it's not that important it works on other systems. On file systems like windows the test should run fine since you don't need to set the execute permission, but I moved the `cfg(unix)` to the top because there is no guarantee that it will work on non unix/windows systems.